sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -1223,13 +1271,16 @@ public Mono<CosmosBatchResponse> executeCosmosBatch(
             requestOptions = new CosmosBatchRequestOptions();
         }
 
-        final CosmosBatchRequestOptions cosmosBatchRequestOptions = requestOptions;
+        CosmosBatchRequestOptions options = batchRequestOptionsAccessor.clone(requestOptions);
```

Comment:
Took a different approach here because the only one where we don't use an internal request option but a public one.